pkg/api/copy_test.go

```diff
@@ -32,24 +32,24 @@ import (
 
 func TestDeepCopyApiObjects(t *testing.T) {
 	for i := 0; i < *fuzzIters; i++ {
-		for _, gv := range []unversioned.GroupVersion{testapi.Default.InternalGroupVersion(), *testapi.Default.GroupVersion()} {
-			f := apitesting.FuzzerFor(t, gv.String(), rand.NewSource(rand.Int63()))
-			for kind := range api.Scheme.KnownTypes(gv) {
-				doDeepCopyTest(t, gv.String(), kind, f)
+		for _, version := range []unversioned.GroupVersion{testapi.Default.InternalGroupVersion(), *testapi.Default.GroupVersion()} {
+			f := apitesting.FuzzerFor(t, version.String(), rand.NewSource(rand.Int63()))
```

Comment:
I think `GroupVersionKind` type name is basically Hungarian notation since its more prescriptive of the individual values it holds than the use case it fulfills.  The contrast in my head is we don't call where your home is a "NumberStreetCityStateZIP", it's an `Address`.

That said, tuple and triplet naming is awkward, and I am not surprised we went down that path since we are mostly trying to clean up existing code that had them as three separate fields.  So I am not against `GroupVersion` or `GroupVersionKind`, but if we create a `GroupVersionKindField`, to describe a field in the API, and start creating `gvkf` acronyms in the code, we have gone too far.

To me, I would think things as follows:
* `group` is a `string`
* `GroupVersion` is more like a `Package` if I were to take Java naming conventions.  This also make sense to me since if we start to support third-party API groups, I would want to add the other fields to a `Package` like vendor for example.
* `kind` is a `Kind`, which is a fully qualified means of addressing an object in a `Package`.  so `kind.Kind` is a `string`, and `kind.Package` is a `Package`.
* `field` is a `Field`, which is a fully qualified means of addressing a member in a `Kind`.  so `field.Field` would be a name we can pass into reflection, etc.

If we are not wanting to describe things in something more akin to the above convention, then I think its confusing to not expand our internal acronyms for folks that come into the code.  I  prefer clarity so `gv` becomes `groupVersion`, `gvk` becomes `groupVersionKind`.